### PR TITLE
Add builder to connection implementation

### PR DESCRIPTION
**Pull request by yaacov**
**Description**

Fixes: #18

### CODE REVIEW

File: pkg/connections/stomp/connection_builder.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	BrokerHost   string
+	BrokerPort   int
+	UserName     string
+	UserPassword string
+	UseTLS       bool
+	InsecureTLS  bool
+}
+
+// Build a new connection object
+func (b *ConnectionBuilder) Build() (c client.Connection, err error) {
```
Comment (zgalor) [inline pkg/connections/stomp/connection_builder.go]: I think the builder shouldn't have a Build() method and we should use:
``` go
func NewConnection(b *Connection) (*ClientConnection, error)

```

That way you can easily call it using:
``` go
c, err := NewConnection(&ConnectionBuilder{
		BrokerHost:   "myHost",
		BrokerPort:   123,
...
})

```
and not having to create a builder instance.
Like the example here:
http://www.theelements.org/golang/2016/05/30/builder-pattern-in-golang.html

Comment (yaacov) [reply pkg/connections/stomp/connection_builder.go]: :+1: Done

File: cmd/messaging-tool/send.go [hunk 2]
```diff
@@ -75,15 +75,16 @@ func runSend(cmd *cobra.Command, args []string) {
 		return
 	}
 
-	c = &stomp.Connection{
+	// Set the clients variables before we can open it.
+	c, _ = stomp.NewConnection(&stomp.ConnectionBuilder{
```
Comment (zgalor) [inline cmd/messaging-tool/send.go]: I know we currently don't return an error but if this is a usage example we should check and handle possible errors

Comment (yaacov) [reply cmd/messaging-tool/send.go]: :+1: Done

**Pull request merged**